include/range/v3/view_interface.hpp

```diff
@@ -297,6 +297,32 @@ namespace ranges
             {
                 return Slice{}(detail::move(derived()), offs.from, offs.to);
             }
+            /// Returns a reference to the element at specified location pos, with bounds checking.
+            template<typename D = Derived,
+                CONCEPT_REQUIRES_(Same<D, Derived>() && RandomAccessRange<D>() && SizedRange<D>())>
+            auto at(range_difference_type_t<D> n) ->
+                decltype(std::declval<D &>().begin()[n])
+            {
+                using size_type = range_size_type_t<Derived>;
+                if (n < 0 || size_type(n) >= ranges::size(derived()))
+                {
+                    throw std::out_of_range("view_interface::at");
+                }
+                return derived().begin()[n];
```

Comment:
I suggest explicit cast of `n` to `size_type` to avoid some `-Wconversion` signedness warnings. The same on L304. (relates #680, gcc would probably stay quiet, but clang would produce a warning)